### PR TITLE
Fix/code smells

### PR DESCRIPTION
**Pull request by CarlosZoft**
[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### Describe your change:

- [ ] Add an algorithm?
- [x] Fix a bug or typo in an existing algorithm?
- [x] Documentation change?

Some minor algorithm fixes

### Checklist:

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [ ] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [ ] All new JavaScript files are placed inside an existing directory.
- [ ] All filenames should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
- [ ] All new algorithms have a URL in their comments that points to Wikipedia or another similar explanation.
- [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: Conversions/DateDayDifference.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ const DateToDay = (dd, mm, yyyy) => {
 
 const DateDayDifference = (date1, date2) => {
   // firstly, check that both input are string or not.
-  if (typeof date1 !== 'string' && typeof date2 !== 'string') {
+  if (typeof date1 !== 'string' || typeof date2 !== 'string') {
```
Comment (CarlosZoft) [inline Conversions/DateDayDifference.js:22]: this should be "or"

File: Maths/LeapYear.js [hunk 2]
```diff
@@ -14,9 +14,5 @@
  * @returns {boolean} true if this is a leap year, false otherwise.
  */
 export const isLeapYear = (year) => {
-  if (year % 400 === 0) return true
-  if (year % 100 === 0) return false
-  if (year % 4 === 0) return true
-
-  return false
+  return ((year % 400) === 0) || (((year % 100) !== 0) && ((year % 4) === 0))
```
Comment (appgurueu) [inline Maths/LeapYear.js:17]: This function seems to be duplicated. Could you deduplicate it?

File: Maths/MidpointIntegration.js [hunk 3]
```diff
@@ -42,7 +42,9 @@ function integralEvaluation (N, a, b, func) {
   // Calculate the integral
   let result = h
   temp = 0
-  for (let i = 0; i < pointsArray.length; i++) temp += pointsArray[i]
+  pointsArray.forEach(point => {
```
Comment (appgurueu) [inline Maths/MidpointIntegration.js:45]: Perhaps even better would be to use [`reduce`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce) to find the sum.

File: Maths/SimpsonIntegration.js [hunk 4]
```diff
@@ -55,7 +55,9 @@ function integralEvaluation (N, a, b, func) {
   // Calculate the integral
   let result = h / 3
   temp = 0
-  for (let i = 0; i < pointsArray.length; i++) temp += pointsArray[i]
+  pointsArray.forEach(point => {
```
Comment (appgurueu) [inline Maths/SimpsonIntegration.js:58]: Same here. Side note: Why is `x => ...` used here but `(v) => ...` used above?

File: Sorts/SimplifiedWiggleSort.js [hunk 5]
```diff
@@ -8,29 +8,30 @@
 import { quickSelectSearch } from '../Search/QuickSelectSearch.js'
 
 export const simplifiedWiggleSort = function (arr) {
+  const arrSize = arr.length
```
Comment (appgurueu) [inline Sorts/SimplifiedWiggleSort.js:11]: I don't think this really helps with readability (or performance).

**Review by appgurueu — DISMISSED** [review]
Definitely an improvement overall. I have some minor comments.

File: Conversions/DateDayDifference.js (hunk 1 above)
Comment (appgurueu) [reply Conversions/DateDayDifference.js:22]: indeed; that's technically a bug fix rather than just refactoring, but it's fine

**Comment by CarlosZoft** [comment]
Ohh nice, tks for review. i'll fix this today xD

**Comment by CarlosZoft** [comment]
@appgurueu, sorry for the delay, fixed the issues ✅

**Review by appgurueu — APPROVED** [review]

**Review by raklaptudirm — APPROVED** [review]

**Review by badr222222 — COMMENTED** [review]
I think the data base was mest up at first but think if there was a few of us we can do it

**Review by badr222222 — COMMENTED** [review]
23 changes: 12 additions & 11 deletions 23  
[Sorts/SimplifiedWiggleSort.js](https://github.com/TheAlgorithms/JavaScript/pull/1338/files/b07f1ddff3d63809ab432a043812113649334ed8#diff-64e8dd4ee35bb747c0862d91ca31466c6f960321258e4cba9e4fd62b02396c22)
 Viewed

@@ -8,29 +8,30 @@
import { quickSelectSearch } from '../Search/QuickSelectSearch.js'

export const simplifiedWiggleSort = function (arr) {
  const arrSize = arr.length
CarlosZoft marked this conversation as resolved.
  // find Median using QuickSelect
  let median = quickSelectSearch(arr, Math.floor(arr.length / 2.0))
  median = median[Math.floor(arr.length / 2.0)]
  let median = quickSelectSearch(arr, Math.floor(arrSize / 2.0))
  median = median[Math.floor(arrSize / 2.0)]

  const sorted = new Array(arr.length)
  const sorted = new Array(arrSize)

  let smallerThanMedianIndx = 0
  let greaterThanMedianIndx = arr.length - 1 - (arr.length % 2)
  let greaterThanMedianIndx = arrSize - 1 - (arrSize % 2)

  for (let i = 0; i < arr.length; i++) {
    if (arr[i] > median) {
      sorted[greaterThanMedianIndx] = arr[i]
  arr.forEach(element => {
    if (element > median) {
      sorted[greaterThanMedianIndx] = element
      greaterThanMedianIndx -= 2
    } else {
      if (smallerThanMedianIndx < arr.length) {
        sorted[smallerThanMedianIndx] = arr[i]
      if (smallerThanMedianIndx < arrSize) {
        sorted[smallerThanMedianIndx] = element
        smallerThanMedianIndx += 2
      } else {
        sorted[greaterThanMedianIndx] = arr[i]
        sorted[greaterThanMedianIndx] = element
        greaterThanMedianIndx -= 2
      }
    }
  }
  })

  return sorted
}
    2 changes: 1 addition & 1 deletion 2  
[String/CheckRearrangePalindrome.js](https://github.com/TheAlgorithms/JavaScript/pull/1338/files/b07f1ddff3d63809ab432a043812113649334ed8#diff-49bd7f329c242bc6a6bef9d83b400333ba0f445f583c0e50a7b196a9db5a6a7f)
 Viewed

@@ -12,7 +12,7 @@ export const palindromeRearranging = (str) => {
    return 'Not a string'
  }
  // Check if is a empty string
  if (str.length === 0) {
  if (!str) {
    return 'Empty string'
  }

Use this

**Pull request merged**